03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,338 @@
+  dplyr::mutate_if(is.numeric, round)
+
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = df_by_var,
+  colData = metadata,
+  design = ~ refinebio_treatment
+)
+```
+
+### 5) Prepare the metadata
+We are going to format the metadata to be used for annotation on the heatmap.
+
+```{r}
+# Let's prepare the metadata for the uncollapsed `DESeqData` set object
+metadata_uncollapsed <- metadata %>%
+  dplyr::select(-c("sample_info_summary", "refinebio_title")) %>%
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+### 6) Define a minimum counts cutoff
+We are going to do some pre-filtering to keep only rows with 10 or more reads
+total.
+
+```{r Define minimum counts cutoff}
+# Let's take a look at the distribution of counts first
+
+# Sum up the counts for each row
+uncollapsed_count_sums <- data.frame(count = rowSums(counts(dds)))
+
+# Create a density plot using the sums
+ggplot2::ggplot(uncollapsed_count_sums, ggplot2::aes(x = count)) +
+  ggplot2::geom_density() +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = 10), # This line represents the current cutoff of 10
+                      color = "blue",
+                      linetype = "dashed") +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = mean(count)), # This line represents the mean count value
+                                   color = "red",
+                                   linetype = "dashed")
+```
+
+```{r}
+# Define a minimum counts cutoff and filter `DESeqDataSet` object to include
+# only rows that have counts above the cutoff
+genes_to_keep <- rowSums(counts(dds)) >= 10
+dds <- dds[genes_to_keep, ]
+```
+
+### 7) Perform DESeq2 normalization
+We are going to use a function from the `DESeq2` package to normalize the data. 
+
+```{r Perform DESeq2 normalization}
+# Normalize the data in the `DESeqDataSet` object using the
+# `normTransform` function from the `DESEq2` R package
+## TODO: Add a note/link to section in documentation on why we use
+# `normTransform` vs `vst` and `rlog`
+ntd_uncollapsed <- normTransform(dds)
+```
+
+### 8) Create a heatmap
+To further customize the heatmap, see a vignette for a guide at this [link](https://slowkow.com/notes/pheatmap-tutorial/).
+
+```{r Create a heatmap}
+# Create and store the heatmap object
+uncollapsed_pheatmap <-
+  pheatmap(
+    assay(ntd_uncollapsed),
+    cluster_rows = FALSE, # Replace with `TRUE` to cluster heatmap by rows (genes in this case)
+    cluster_cols = FALSE, # Replace with `TRUE` to cluster heatmap by columns (samples in this case)
+    annotation_col = metadata_uncollapsed,
+    main = "Technical Replicates Heatmap",
+    colorRampPalette(c("deepskyblue",
+                       "black",
+                       "yellow"))(25
+                       )
+  )
+```
+
+"Now that we've seen our replicates are fairly identical... let's take a look at our experimental question". 
+
+## How are our biological replicates clustering together?
+
+### 4) Create a DESeqDataset
```

Comment:
We may want to get rid of the numbering? What do you think?

Reply:
I had the same though as it could be a bit confusing, although I also do think that it can be helpful in allowing users to keep track of the steps sequentially if they go back to the notebook at some point and want to use it to answer just one of the questions.

Reply:
Can we go back and make it 4a and 4b or is that more confusing?